utils/gql.py

```diff
@@ -2,34 +2,83 @@
 import requests
 import os
 import contextlib
+import textwrap
+import logging
 
 from graphqlclient import GraphQLClient
 from utils.config import get_config
 
 _gqlapi = None
 
 
+INTEGRATIONS_QUERY = """
+{
+    integrations: integrations_v1 {
+        name
+        schemas
+    }
+}
+"""
+
+
 class GqlApiError(Exception):
     pass
 
 
+class GqlApiIntegrationNotFound(Exception):
+    def __init__(self, integration):
+        msg = f"""
+        Integration not found: {integration}
+
+        Integration should be defined in App-Interface with the
+        /app-sre/integration-1.yml schema.
+        """
+        super().__init__(textwrap.dedent(msg).strip())
+
+
+class GqlApiErrorForbiddenSchema(Exception):
+    def __init__(self, schema):
+        msg = f"""
+        Forbidden schema: {schema}
+
+        The `schemas` parameter in the integration file in App-Interface
+        should be updated to include this schema.
+        """
+        super().__init__(textwrap.dedent(msg).strip())
+
+
 class GqlGetResourceError(Exception):
     def __init__(self, path, msg):
         super(GqlGetResourceError, self).__init__(
-            "error getting resource from path {}: {}".format(path, str(msg))
+            "Error getting resource from path {}: {}".format(path, str(msg))
         )
 
 
 class GqlApi(object):
-    def __init__(self, url, token=None):
+    _valid_schemas = None
+
+    def __init__(self, url, token=None, int_name=None, validate_schemas=False):
         self.url = url
         self.token = token
+        self.integration = int_name
+        self.validate_schemas = validate_schemas
```

Comment:
maybe we can do:
```
self.validate_schemas = validate_schemas and int_name
```

Reply:
I like it!

Reply:
well, on second thought this is not as easy, I'd rather raise an exception if you try to validate schemas without definiing the integration name?